src/sacn/mem.c

```diff
@@ -35,6 +37,72 @@
+  do                                                                                                  \
+  {                                                                                                   \
+    etcpal_rbtree_clear_with_cb(&((SacnSourceUniverse*)ptr)->unicast_dests, free_unicast_dests_node); \
+    if (((SacnSourceUniverse*)ptr)->netints)                                                          \
+      free(((SacnSourceUniverse*)ptr)->netints);                                                      \
+    free(ptr);                                                                                        \
+  } while (0)
+#define ALLOC_SACN_SOURCE_NETINT() malloc(sizeof(SacnSourceNetint))
+#define FREE_SACN_SOURCE_NETINT(ptr) free(ptr)
+#define ALLOC_SACN_UNICAST_DESTINATION() malloc(sizeof(SacnUnicastDestination))
+#define FREE_SACN_UNICAST_DESTINATION(ptr) free(ptr)
+#define ALLOC_SACN_SOURCE_RB_NODE() malloc(sizeof(EtcPalRbNode))
+#define FREE_SACN_SOURCE_RB_NODE(ptr) free(ptr)
+#elif SACN_SOURCE_ENABLED
+#define ALLOC_SACN_SOURCE() etcpal_mempool_alloc(sacnsource_source_states)
+#define FREE_SACN_SOURCE(ptr)                                                         \
+  do                                                                                  \
+  {                                                                                   \
+    etcpal_rbtree_clear_with_cb(&((SacnSource*)ptr)->universes, free_universes_node); \
+    etcpal_rbtree_clear_with_cb(&((SacnSource*)ptr)->netints, free_netints_node);     \
+    etcpal_mempool_free(sacnsource_source_states, ptr);                               \
+  } while (0)
+#define ALLOC_SACN_SOURCE_UNIVERSE() etcpal_mempool_alloc(sacnsource_universe_states)
+#define FREE_SACN_SOURCE_UNIVERSE(ptr)                                                                \
+  do                                                                                                  \
+  {                                                                                                   \
+    etcpal_rbtree_clear_with_cb(&((SacnSourceUniverse*)ptr)->unicast_dests, free_unicast_dests_node); \
+    etcpal_mempool_free(sacnsource_universe_states, ptr);                                             \
+  } while (0)
+#define ALLOC_SACN_SOURCE_NETINT() etcpal_mempool_alloc(sacnsource_netints)
+#define FREE_SACN_SOURCE_NETINT(ptr) etcpal_mempool_free(sacnsource_netints, ptr)
+#if SACN_SOURCE_UNICAST_ENABLED
+#define ALLOC_SACN_UNICAST_DESTINATION() etcpal_mempool_alloc(sacnsource_unicast_dests)
+#define FREE_SACN_UNICAST_DESTINATION(ptr) etcpal_mempool_free(sacnsource_unicast_dests, ptr)
+#else
+#define ALLOC_SACN_UNICAST_DESTINATION() NULL
+#define FREE_SACN_UNICAST_DESTINATION(ptr)
+#endif
+#define ALLOC_SACN_SOURCE_RB_NODE() etcpal_mempool_alloc(sacnsource_rb_nodes)
+#define FREE_SACN_SOURCE_RB_NODE(ptr) etcpal_mempool_free(sacnsource_rb_nodes, ptr)
+#else
```

Comment:
This particular block of code no longer exists, but I did try to apply this in other areas where it could be helpful. See https://github.com/ETCLabs/sACN/pull/32/commits/a49e66e65071ffff22827c2004d033b07dadf149. I didn't comment the ones that had enough clarity already - meaning where there was no nesting and the corresponding if was close enough. But if you see more areas that should have a comment let me know. Resolving for now.